client/pages/results.jsx

```diff
@@ -61,23 +67,101 @@ export default class Results extends React.Component {
     return author.join(', ');
   }
 
+  handleSavedItem(target) {
+    const { results } = this.state;
+    const books = results.items;
+    for (let i = 0; i < books.length; i++) {
+      if (books[i].id === target.id) {
+        const info = {
+          title: books[i].volumeInfo.title,
+          googleId: books[i].id,
+          coverUrl: (books[i].volumeInfo.imageLinks) ? books[i].volumeInfo.imageLinks.thumbnail : null,
+          author: this.handleAuthor(books[i].volumeInfo.authors)
+        };
+        return info;
+      }
+    }
+  }
+
+  handleSave(event) {
+    const target = event.target;
+    const req = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(this.handleSavedItem(target))
+    };
+    fetch('/api/bookShelf', req)
+      .then(res => res.json())
+      .then(result => {
+        if (result.error) {
+          this.setState({
+            isError: true
+          });
+          setTimeout(() => {
+            this.setState({
+              isError: false
+            });
+          }, 3000);
+        } else {
+          this.setState({
+            isSaved: true
+          });
+          setTimeout(() => {
+            this.setState({
+              isSaved: false
+            });
+          }, 3000);
+        }
+      })
+      .catch(error => console.error(error));
+
+  }
+
+  handleHeading() {
```

Comment:
You have a lot of methods named `handle...`. This is a bit confusing. `handle` is usually used to denote "event handlers". If your method or function is supposed to "render" something or "get" something, use that instead.

- `renderHeading`
- `getAuthors`
- `getSavedItem`